### ISSUE TITLE
[Bug] bladebit cudaplot produces invalid plots

### ISSUE THREAD

**Issue by MichaelFolder** [Chia-Network/chia-blockchain#16405]
### What happened?

.\bladebit\bladebit_cuda cudaplot makes the invalid plots. .\chia plots check -n 30|100 -g paths returns 0/30 or 0/100 Proofs for this, and warnings that this plots are INVALID. I checked different sequences of parameters, long and short versions of their names, different compressions (c1, c5, c7), the result is the same. GUI-plotting does the same result.

Below is the log for creating plot on a 256GB memory and hard drive with a check then. Usually I create plots on an intermediate RAM-disk, it's faster, and there are no problems with [PlotWriter], then copy from there to the target disk with standalone Windows file copy procedure, but the result is the same. There are no such problems with other plotters that run on memory or RAM-disks/SSD, including bladebit 1.2.4.

Platform:
2x Xeon E5-2690v4 on Supermicro X10DRi-LN4F+
896Gb RAM (512Gb on CPU1, 384Gb on CPU2),
NUMA feature is disabled,
Windows 10 Pro for Workstations

### Version

2.0.1

### What platform are you using?

Windows

### What ui mode are you using?

CLI

### Relevant log output

```shell
PS C:\Program Files\Chia\resources\app.asar.unpacked\daemon> bladebit\bladebit_cuda -t 24 -z 1 -f 8d15fc9dbe8d260e923951a6a63af6eb51801145ce5df343dcf7711729640645a1c56e400b7da699af81b8ce4d349490 -c xch1j32nlt3ezevhc7slm6cakyn5wlrpv4ue8vhx8hmdlwg2sxcc2srslxhsvj -m cudaplot W:\chiasea6-8\

Bladebit Chia Plotter
Version      : 3.0.0
Git Commit   : 02a8e68dcf0e971c72fc8377b89cdd081b80b843
Compiled With: msvc 19.29.30151

[Global Plotting Config]
 Will create 1 plots.
 Thread count          : 24
 Warm start enabled    : false
 NUMA disabled         : true
 CPU affinity disabled : false
 Farmer public key     : 8d15fc9dbe8d260e923951a6a63af6eb51801145ce5df343dcf7711729640645a1c56e400b7da699af81b8ce4d349490
 Pool contract address : xch1j32nlt3ezevhc7slm6cakyn5wlrpv4ue8vhx8hmdlwg2sxcc2srslxhsvj
 Compression Level     : 1
 Benchmark mode        : disabled

[Bladebit CUDA Plotter]
Selected cuda device 0 : NVIDIA GeForce RTX 3060
 CUDA Compute Capability   : 8.6
 SM count                  : 28
 Max blocks per SM         : 16
 Max threads per SM        : 1536
 Async Engine Count        : 2
 L2 cache size             : 2.25 MB
 L2 persist cache max size : 1.69 MB
 Stack Size                : 1.00 KB
 Memory:
  Total                    : 12.00 GB
  Free                     : 9.96 GB

Allocating buffers (this may take a few seconds)...
Kernel RAM required       : 90240524288  bytes ( 86060.07  MiB or 84.04  GiB )
Intermediate RAM required : 2999001088   bytes ( 2860.07   MiB or 2.79   GiB )
Host RAM required         : 141733920768 bytes ( 135168.00 MiB or 132.00 GiB )
Total Host RAM required   : 231974445056 bytes ( 221228.07 MiB or 216.04 GiB )
GPU RAM required          : 6140243968   bytes ( 5855.79   MiB or 5.72   GiB )
Allocating buffers

Generating plot 1 / 1: 6de7a4da64c9b356cc7a7753644b79196186cad290a7f29e2727dff2981b63dd
Plot temporary file: W:\chiasea6-8\plot-k32-c01-2023-09-24-12-13-6de7a4da64c9b356cc7a7753644b79196186cad290a7f29e2727dff2981b63dd.plot.tmp

Generating F1
Finished F1 in 2.92 seconds.
Table 2 completed in 17.12 seconds with 4294967296 entries.
Table 3 completed in 18.42 seconds with 4294876243 entries.
Table 4 completed in 50.42 seconds with 4294830039 entries.
Table 5 completed in 47.52 seconds with 4294546251 entries.
Table 6 completed in 43.32 seconds with 4294129381 entries.
Table 7 completed in 38.75 seconds with 4293215375 entries.
Finalizing Table 7
Finalized Table 7 in 19.14 seconds.
Completed Phase 1 in 238.09 seconds
Marked Table 6 in 9.19 seconds.
Marked Table 5 in 8.93 seconds.
Marked Table 4 in 8.63 seconds.
Marked Table 3 in 11.71 seconds.
Completed Phase 2 in 38.48 seconds
Compressing Table 2 and 3...
 Step 1 completed step in 13.67 seconds.
 Step 2 completed step in 10.92 seconds.
Completed table 2 in 24.58 seconds with 3439753444 / 4294876243 entries ( 80.09% ).
Compressing tables 3 and 4...
 Step 1 completed step in 8.23 seconds.
 Step 2 completed step in 7.03 seconds.
 Step 3 completed step in 11.08 seconds.
Completed table 3 in 26.34 seconds with 3465829499 / 4294830039 entries ( 80.70% ).
Compressing tables 4 and 5...
 Step 1 completed step in 11.00 seconds.
 Step 2 completed step in 7.13 seconds.
[PlotWriter] Command buffer full. Waiting for commands.
[PlotWriter] Waited 0.000000 seconds for a Command to be available.
[PlotWriter] Command buffer full. Waiting for commands.
[PlotWriter] Waited 12.432000 seconds for a Command to be available.
 Step 3 completed step in 23.82 seconds.
Completed table 4 in 41.96 seconds with 3532469085 / 4294546251 entries ( 82.25% ).
Compressing tables 5 and 6...
 Step 1 completed step in 10.84 seconds.
 Step 2 completed step in 7.40 seconds.
[PlotWriter] Command buffer full. Waiting for commands.
[PlotWriter] Waited 1.448000 seconds for a Command to be available.
[PlotWriter] Command buffer full. Waiting for commands.
[PlotWriter] Waited 21.011000 seconds for a Command to be available.
[PlotWriter] Command buffer full. Waiting for commands.
[PlotWriter] Waited 20.213000 seconds for a Command to be available.
[PlotWriter] Command buffer full. Waiting for commands.
[PlotWriter] Waited 20.111000 seconds for a Command to be available.
 Step 3 completed step in 76.73 seconds.
Completed table 5 in 94.97 seconds with 3712720379 / 4294129381 entries ( 86.46% ).
Compressing tables 6 and 7...
 Step 1 completed step in 10.97 seconds.
 Step 2 completed step in 8.20 seconds.
[PlotWriter] Command buffer full. Waiting for commands.
[PlotWriter] Waited 0.000000 seconds for a Command to be available.
[PlotWriter] Command buffer full. Waiting for commands.
[PlotWriter] Waited 20.543000 seconds for a Command to be available.
[PlotWriter] Command buffer full. Waiting for commands.
[PlotWriter] Waited 20.862000 seconds for a Command to be available.
[PlotWriter] Command buffer full. Waiting for commands.
[PlotWriter] Waited 21.505000 seconds for a Command to be available.
 Step 3 completed step in 78.52 seconds.
Completed table 6 in 97.69 seconds with 4293215375 / 4293215375 entries ( 100.00% ).
Serializing P7 entries
[PlotWriter] Command buffer full. Waiting for commands.
[PlotWriter] Waited 20.767000 seconds for a Command to be available.
[PlotWriter] Command buffer full. Waiting for commands.
[PlotWriter] Waited 23.960000 seconds for a Command to be available.
Completed serializing P7 entries in 48.97 seconds.
Completed Phase 3 in 334.54 seconds
Completed Plot 1 in 611.12 seconds ( 10.19 minutes )

W:\chiasea6-8\plot-k32-c01-2023-09-24-12-13-6de7a4da64c9b356cc7a7753644b79196186cad290a7f29e2727dff2981b63dd.plot.tmp -> W:\chiasea6-8\plot-k32-c01-2023-09-24-12-13-6de7a4da64c9b356cc7a7753644b79196186cad290a7f29e2727dff2981b63dd.plot
Completed writing plot in 308.28 seconds
Final plot table pointers:
 Table 1:                0 ( 0x0000000000000000 )
 Table 2:       1289687564 ( 0x000000004cdf120c )
 Table 3:      15290566412 ( 0x000000038f63870c )
 Table 4:      29378963912 ( 0x00000006d71f69c8 )
 Table 5:      43738248587 ( 0x0000000a2f00b58b )
 Table 6:      58830241487 ( 0x0000000db28e3ecf )
 Table 7:      76281914012 ( 0x00000011c2c1a29c )
 C 1    :             4096 ( 0x0000000000001000 )
 C 2    :          1721388 ( 0x00000000001a442c )
 C 3    :          1721564 ( 0x00000000001a44dc )

Final plot table sizes:
 Table 1: 0.00 MiB
 Table 2: 13352.28 MiB
 Table 3: 13435.74 MiB
 Table 4: 13694.08 MiB
 Table 5: 14392.85 MiB
 Table 6: 16643.21 MiB
 Table 7: 16889.11 MiB
 C 1    : 1.64 MiB
 C 2    : 0.00 MiB
 C 3    : 1228.30 MiB


PS C:\Program Files\Chia\resources\app.asar.unpacked\daemon> .\chia plots check -n 100 -g W:\
2023-09-24T12:45:34.547  chia.plotting.check_plots        : INFO     Loading plots in config.yaml using plot_manager loading code (parallel read: True)

...

2023-09-24T12:45:36.450  chia.plotting.check_plots        : INFO     Testing plot W:\chiasea6-8\plot-k32-c01-2023-09-24-12-13-6de7a4da64c9b356cc7a7753644b79196186cad290a7f29e2727dff2981b63dd.plot k=32
2023-09-24T12:45:36.451  chia.plotting.check_plots        : INFO        Pool contract address:  xch1j32nlt3ezevhc7slm6cakyn5wlrpv4ue8vhx8hmdlwg2sxcc2srslxhsvj
2023-09-24T12:45:36.452  chia.plotting.check_plots        : INFO        Farmer public key:      8d15fc9dbe8d260e923951a6a63af6eb51801145ce5df343dcf7711729640645a1c56e400b7da699af81b8ce4d349490

...

2023-09-24T12:45:36.531  chia.plotting.check_plots        : ERROR       Proofs 0 / 100, 0.0 Filepath: W:\chiasea6-8\plot-k32-c01-2023-09-24-12-13-6de7a4da64c9b356cc7a7753644b79196186cad290a7f29e2727dff2981b63dd.plot

...

2023-09-24T12:45:36.578  chia.plotting.check_plots        : ERROR       Proofs 0 / 100, 0.0 Filepath: W:\chiasea6-8\plot-k32-c05-2023-09-24-11-15-8c93aa7f4b24c76d8d4460eab4b1308f74d6edbb09e22a2cb431736b23fa27da.plot
2023-09-24T12:45:36.578  chia.plotting.check_plots        : INFO
2023-09-24T12:45:36.578  chia.plotting.check_plots        : INFO
2023-09-24T12:45:36.578  chia.plotting.check_plots        : INFO     Summary
2023-09-24T12:45:36.578  chia.plotting.check_plots        : INFO     Found 0 valid plots, total size 0.00000 TiB
2023-09-24T12:45:36.578  chia.plotting.check_plots        : WARNING  4 invalid plots found:
2023-09-24T12:45:36.578  chia.plotting.check_plots        : WARNING      4 bad plots:
2023-09-24T12:45:36.593  chia.plotting.check_plots        : WARNING  W:\chiasea6-8\plot-k32-c01-2023-09-24-11-58-44ab82c58c04014abc722f69227989fd9d540649a03d0ffc112f90f8e44fe437.plot
2023-09-24T12:45:36.593  chia.plotting.check_plots        : WARNING  W:\chiasea6-8\plot-k32-c01-2023-09-24-12-13-6de7a4da64c9b356cc7a7753644b79196186cad290a7f29e2727dff2981b63dd.plot
2023-09-24T12:45:36.593  chia.plotting.check_plots        : WARNING  W:\chiasea6-8\plot-k32-c05-2023-09-24-11-08-d13588e764e2fe51017616dc3bc9fde1bc55e6efa474bb880acc5b63023c0c17.plot
2023-09-24T12:45:36.593  chia.plotting.check_plots        : WARNING  W:\chiasea6-8\plot-k32-c05-2023-09-24-11-15-8c93aa7f4b24c76d8d4460eab4b1308f74d6edbb09e22a2cb431736b23fa27da.plot
PS C:\Program Files\Chia\resources\app.asar.unpacked\daemon>
```

**Comment by shaneo257** [comment]
Hi,

It is a known issue with Windows 10 and Bladebit Cuda creating invalid plots, the current work around is to use either Win 11, or Linux

**Comment by emlowe** [comment]
The next release fixes issues with Windows 10 - but the 3.0.0 version does not work on Windows 10 - this is a known issue
